backend/fms_core/services/project.py

```diff
@@ -39,4 +42,57 @@ def create_project(name=None, principal_investigator=None, requestor_name=None,
     except ValidationError as e:
         errors.append(str(e))
 
-    return (project, errors, warnings)
\ No newline at end of file
+    return (project, errors, warnings)
+
+def add_sample_to_study(sample: Sample, project: Project, study_letter: str, step_order: int | None = None) -> tuple[dict[str, list | str], dict[str, list | str]]:
+    """Add a sample to a study of a project.
+    If `step_order` it is not specified, the sample will be queued at the start of the study.
+
+    Args:
+        sample: Sample instance
+        project: Project instance
+        study_letter: Letter of an existing study
+        step_order: Step order of a step in the study. Defaults to None.
+
+    Returns:
+        Tuple of errors and warnings
+    """
+
+    errors = defaultdict(list)
+    warnings = defaultdict(list)
+
+    def add_sample_to_study_(sample: Sample, project: Project, study_letter: str, step_order: int | None = None):
```

Comment:
Why make an internal call here to a function is everything in that service ? It looks silly and makes no sense to me ...

Reply:
It seems to also break the previous way of dealing with the errors in the template (it could previously report more than one error by row ...). I think you treat this like javascript while this is not the function inside does not have access to the same errors as the external one ... the error management you do here is likely not working in python ...